examples/repository/example_test.go

```diff
@@ -0,0 +1,148 @@
+package repository_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/alpine-hodler/gidari/proto"
+	"github.com/alpine-hodler/gidari/repository"
+	"github.com/alpine-hodler/gidari/tools"
+	_ "github.com/joho/godotenv/autoload"
+)
+
+func ExampleNew() {
+	// cluster_uri looks something like 'mongodb+srv://<username>:<password>@<clustername><host>/<Collection>'
+	dsn := os.Getenv("MONGO_URI")
+	ctx := context.TODO()
+
+	repo, err := repository.New(ctx, dsn)
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println(repo.Storage.IsNoSQL())
+	// Output:
+	// true
+}
+
+// TODO
+// Didn't really know how to make a meaningful example for NewTx(), but heres some boilerplate to get statrted
+
+// func ExampleNewTx() {
+// 	dsn := os.Getenv("MONGO_URI")
+// 	ctx := context.TODO()
+//
+// 	txRepo, err := repository.NewTx(ctx, dsn)
+// 	if err != nil {
+// 		panic(err)
+// 	}
+//
+// 	txInit, err := txRepo.StartTx(ctx)
+//
+// 	if condition {
+// 		txInit.Rollback()
+// 	}
+// 	txInit.Commit()
```

Comment:
This is great! I think if we just tweak it like so it will be a good example:

```go
req := &proto.UpsertRequest{
    Table:    "TxnExampleTable",
    Data:     []byte(`[{"id": "7fd0abc0-e5ad-4cbb-8d54-f2b3f43364da"}]`),
    DataType: int32(tools.UpsertDataJSON),
}

rsp, err := txRepo.Upsert(ctx, req)
if err != nil {
    panic(err)
}

if err := txRepo.Commit(); err != nil {
    panic(err)
}
```

We can drop this line since `NewTx` does all of this under the hood:

```go
txInit, err := txRepo.StartTx(ctx)
```